src/WrappedMToken.sol

```diff
@@ -706,8 +724,21 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
      * @return unwrapped_ The amount of M withdrawn.
      */
     function _unwrap(address account_, address recipient_, uint240 amount_) internal returns (uint240 unwrapped_) {
-        _burn(account_, amount_);
-        _transferM(recipient_, unwrapped_ = amount_);
+        _burn(account_, unwrapped_ = amount_);
+
+        uint240 startingBalance_ = _mBalanceOf(address(this));
+
+        // NOTE: The behavior of `IMTokenLike.transfer` is known, so its return can be ignored.
+        IMTokenLike(mToken).transfer(recipient_, amount_);
+
+        // NOTE: When this WrappedMToken contract is earning, any amount of M sent from it is converted to a principal
+        //       amount at the MToken contract, which when represented as a present amount, may be a rounding error
+        //       amount more than `amount_`. In order to capture the real decrease in M, the difference between the
+        //       ending and starting M balance is captured.
+        uint240 decrease_ = startingBalance_ - _mBalanceOf(address(this));
+
+        // If the M lost is more than the wM burned, then the difference is added to `roundingError`.
+        roundingError += uint144(decrease_ - amount_);
```

Comment:
So `decrease_` can never be lower than `amount_`?
Would this still be the case if the Wrapper stops earning?

Reply:
All of this makes starts with the premise that the wrapper is earning.